flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCUpsertOutputFormatTest.java

```diff
@@ -66,6 +67,7 @@ public void testJDBCOutputFormat() throws Exception {
 						.build())
 				.setFieldNames(fieldNames)
 				.setKeyFields(keyFields)
+				.setFieldTypes(SQL_TYPES)
```

Comment:
It seems that it is error-prone if we don't set a field types? If so, I would suggest to require field types in the JDBCUpsertOutputFormat.Builder. As it is a new introduced class, it's fine to change this.